torchtext/datasets/language_modeling.py

```diff
@@ -1,217 +1,199 @@
-                load the validation set. Default: 'wiki.valid.tokens'.
-            test: The filename of the test data, or None to not load the test
-                set. Default: 'wiki.test.tokens'.
-        """
-        return super(WikiText2, cls).splits(
-            root=root, train=train, validation=validation, test=test,
-            text_field=text_field, **kwargs)
-
-    @classmethod
-    def iters(cls, batch_size=32, bptt_len=35, device=0, root='.data',
-              vectors=None, **kwargs):
-        """Create iterator objects for splits of the WikiText-2 dataset.
-
-        This is the simplest way to use the dataset, and assumes common
-        defaults for field, vocabulary, and iterator parameters.
+            data: a tensor of tokens. tokens are ids after
+                numericalizing the string tokens.
+                torch.Tensor([token_id_1, token_id_2, token_id_3, token_id1]).long()
+            vocab: Vocabulary object used for dataset.
+
+        Examples:
+            >>> from torchtext.vocab import build_vocab_from_iterator
+            >>> data = torch.Tensor([token_id_1, token_id_2,
+                                     token_id_3, token_id_1]).long()
+            >>> vocab = build_vocab_from_iterator([['language', 'modeling']])
+            >>> dataset = LanguageModelingDataset(data, vocab)
 
-        Arguments:
-            batch_size: Batch size.
-            bptt_len: Length of sequences for backpropagation through time.
-            device: Device to create batches on. Use -1 for CPU and None for
-                the currently active GPU device.
-            root: The root directory that the dataset's zip archive will be
-                expanded into; therefore the directory in whose wikitext-2
-                subdirectory the data files will be stored.
-            wv_dir, wv_type, wv_dim: Passed to the Vocab constructor for the
-                text field. The word vectors are accessible as
-                train.dataset.fields['text'].vocab.vectors.
-            Remaining keyword arguments: Passed to the splits method.
         """
-        TEXT = data.Field()
-
-        train, val, test = cls.splits(TEXT, root=root, **kwargs)
-
-        TEXT.build_vocab(train, vectors=vectors)
-
-        return data.BPTTIterator.splits(
-            (train, val, test), batch_size=batch_size, bptt_len=bptt_len,
-            device=device)
 
+        super(LanguageModelingDataset, self).__init__()
+        self._data = data
+        self._vocab = vocab
+
+    def __getitem__(self, i):
+        return self._data[i]
+
+    def __len__(self):
+        return len(self._data)
+
+    def __iter__(self):
+        for x in self._data:
+            yield x
+
+    def get_vocab(self):
+        return self._vocab
+
+
+def _setup_datasets(dataset_name, tokenizer=get_tokenizer("basic_english"),
+                    root='.data', vocab=None, removed_tokens=['<unk>']):
+    if dataset_name == 'PennTreebank':
```

Comment:
If we use that function for future datasets, is there a more general strategy than just making `PennTreebank` a special case?